syntax = "proto3";

package libp2p.peer.persistent.pb;

import "libp2p/peer/pb/crypto.proto";
import "libp2p/peer/pb/envelope.proto";

option go_package = "github.com/libp2p/py-libp2p/libp2p/peer/persistent/pb";

// PeerAddresses contains a list of multiaddresses for a peer
message PeerAddresses {
    repeated bytes addresses = 1;  // Each address is a serialized multiaddr
}

// PeerProtocols contains a list of supported protocols for a peer
message PeerProtocols {
    repeated string protocols = 1;
}

// PeerMetadata contains key-value metadata for a peer
message PeerMetadata {
    map<string, bytes> metadata = 1;
}

// PeerKeys contains the public and private keys for a peer
message PeerKeys {
    libp2p.peer.pb.crypto.PublicKey public_key = 1;
    libp2p.peer.pb.crypto.PrivateKey private_key = 2;
}

// PeerLatency contains latency information for a peer
message PeerLatency {
    int64 latency_ns = 1;  // Latency in nanoseconds
}

// PeerEnvelope contains a signed envelope for a peer
message PeerEnvelope {
    libp2p.peer.pb.record.Envelope envelope = 1;
}

// PeerRecordState contains the state of a peer record
message PeerRecordState {
    enum State {
        UNKNOWN = 0;
        VALID = 1;
        INVALID = 2;
        EXPIRED = 3;
    }
    State state = 1;
}
